lib/rucio/web/ui/static/did.js

```diff
@@ -181,6 +181,63 @@ load_replicas = function(scope, name) {
     });
 };
 
+download_file = function(link, file_scope, filename, rse, filesize, dataset_scope, dataset) {
+    if(typeof(uuid) === 'undefined') {
+        console.log('UUID not found, no trace sent.');
+        return;
+    }
+    var options = {
+        eventVersion: 'webui_' + get_version(),
+        protocol: 'davs',
+        uuid: uuid.replace(/-/g,''),
+        datasetScope: dataset_scope,
+        eventType: 'download',
+        remoteSite: rse,
+        dataset: dataset,
+        filename: filename,
+        filesize: filesize,
+        scope: file_scope,
+        clientState: 'DONE',
+        success: function(data) {},
+        error: function(jqXHR, textStatus, errorThrown) {
+            console.log(jqXHR);
+        }
+    };
+    r.send_trace(options);
+    var element = document.createElement('a');
+    element.setAttribute('href', link);
+    element.setAttribute('target', '_blank');
+    element.setAttribute('download', filename);
+    element.style.display = 'none';
+    document.body.appendChild(element);
+    element.click();
+    document.body.removeChild(element);
+};
+
+uuid4 = function () {
+    var uuid = '';
+    for (var i = 0; i < 32; i ++) {
+        switch (i) {
+            case 8:
+            case 20:
+                uuid += '-';
```

Comment:
As you are removing the dashes later on anyway, why not just omitting them here in the first place?

Reply:
To have a function that meets the RFC4122 requirements - but I can omit them here already if you think thats better :)

Reply:
OK I see your point. But here it's not a utility function which will be used anywhere else but rather just there to create the uuid for the traces, which are without dashes.